### PR TITLE
Add documentation for bytes formatting error code

### PR DESCRIPTION
**Pull request by hauntsaninja**
See #14959

### CODE REVIEW

File: docs/source/error_code_list.rst [hunk 1]
```diff
@@ -908,6 +908,24 @@ Functions will always evaluate to true in boolean contexts.
     if f:  # Error: Function "Callable[[], Any]" could always be true in boolean context  [truthy-function]
         pass
 
+Check that function isn't used in boolean context [str-bytes-safe]
```
Comment (JelleZijlstra) [inline docs/source/error_code_list.rst:911]:
```suggestion
Check for unexpected bytes coercions [str-bytes-safe]
```

File: docs/source/error_code_list.rst [hunk 2]
```diff
@@ -908,6 +908,24 @@ Functions will always evaluate to true in boolean contexts.
...
+-------------------------------------------------------------------
+
+Warn about dangerous coercions related to bytes and string types.
```
Comment (JelleZijlstra) [inline docs/source/error_code_list.rst:914]: Not sure "dangerous" is the right word since it won't crash, just give potentially unexpected output.

Comment (ikonst) [reply docs/source/error_code_list.rst:914]: I feel like the theme here is "implicit" which makes the intention ambiguous.

Comment (JelleZijlstra) [reply docs/source/error_code_list.rst:914]: Agree, "implicit" is a good word to use

**Review by ethanhs — APPROVED** [review]

**Pull request merged**